frontend/src/components/data-table/DataTable.tsx

```diff
@@ -19,6 +23,19 @@ const DataTable: React.FC<IDataTableProps> = (props) => {
   const { contract } = props;
   const tokenName = process.env.REACT_APP_PAYMENT_TOKEN_NAME || '';
 
+  const wallet = useWallet();
+  const isSupplier =
+    (wallet.account || '').toLowerCase() ===
+    (contract.supplier || '').toLowerCase();
+
+  // determine buttons
+  let button = <></>;
+  if (isSupplier && contract.state === JobState.Available) {
+    button = <CancelJobButton />;
+  } else if (contract.state === JobState.Available) {
+    button = <StartJobButton />;
+  }
```

Comment:
This is OK for now, but we might want to create a `<ContractButton />` component that encapsulates all this logic. There will be more "button types" coming. It's ok though, I'll make an issue, let's just note that for an improvement and get this merged in.